apps/api/src/user/service/user.service.ts

```diff
@@ -32,6 +41,28 @@ export class UserService {
       profilePictureUrl: dto?.profilePictureUrl,
       isOnboardingFinished: dto.isOnboardingFinished
     }
+    if (dto?.email) {
+      const userwithEmail = await this.prisma.user.findFirst({
+        where: {
+          email: dto.email
+        }
+      })
+
+      if (userwithEmail) {
+        throw new ConflictException('User with this email already exists')
+      }
+
+      const otp = await this.prisma.userEmailChange.create({
+        data: {
+          newEmail: dto.email,
+          userId: user.id,
+          otp: randomUUID().slice(0, 6).toUpperCase()
+        }
+      })
+
+      await this.mailService.sendOtp(dto.email, otp.otp)
```

Comment:
You would need to add another function to the mail service for this specific function. This is so because we will be using different templates. You can create a function named `sendEmailChangedOtp`

Reply:
Added